File: app/src/main/java/com/demo/adapter/ClickActivity.java
package com.demo.adapter;

import android.os.Bundle;
import android.util.Log;
import android.view.View;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import lib.kalu.adapter.BaseCommonAdapter;
import lib.kalu.adapter.decoration.SpaceDecoration;
import lib.kalu.adapter.holder.RecyclerHolder;

public class ClickActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_click);

        final ArrayList<String> list = new ArrayList<>();
        for (int i = 0; i < 20; i++) {
            list.add(String.valueOf(i));
        }

        final int[] i = {1};

        BaseCommonAdapter<String> adapter = new BaseCommonAdapter<String>() {

            @Override
            protected int onView() {
                return R.layout.activity_click_item;
            }

            @NonNull
            @Override
            protected List<String> onData() {
                return list;
            }

            @Override
            protected void onHolder(final RecyclerView.LayoutManager manager, final RecyclerHolder holder, int type) {

                Log.e("holder", "onHolder = i = "+ i[0]);
                ++i[0];

                holder.getView(R.id.click_text1).setOnClickListener(new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {
                        int position = manager.getPosition(holder.itemView);
                        Log.e("holder", "position = " + position);
                    }
                });
            }

            @Override
            protected void onNext(RecyclerHolder holder, String model, int position) {
                Log.e("kalu", "position = " + position);
            }

            @Override
            protected void onScroll(int horizontalOffset, int verticalOffset) {
                super.onScroll(horizontalOffset, verticalOffset);
            }
        };

        RecyclerView view = findViewById(R.id.list);
        view.addItemDecoration(new SpaceDecoration(10f));
        view.setLayoutManager(new LinearLayoutManager(getApplicationContext()));
        view.setAdapter(adapter);
    }
}

File: app/src/main/java/com/demo/adapter/ClickActivity.kt
package com.demo.adapter

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.RecyclerView
import android.view.View
import android.widget.Toast
import lib.kalu.adapter.BaseCommonAdapter
import lib.kalu.adapter.decoration.SpaceDecoration
import lib.kalu.adapter.holder.RecyclerHolder
import lib.kalu.adapter.manager.CrashGridLayoutManager
import java.util.*

class ClickActivity : AppCompatActivity() {

    private val mRecyclerView: RecyclerView? by lazy {
        findViewById(R.id.click_recycler) as RecyclerView
    }

    private val mArrayList: ArrayList<String>? by lazy {
        ArrayList<String>()
    }

    private val mLayoutManager: CrashGridLayoutManager? by lazy {
        CrashGridLayoutManager(applicationContext, 1)
    }

    private val mSpaceDecoration: SpaceDecoration? by lazy {
        SpaceDecoration(10f)
    }

    private val mLoadAdapter: BaseCommonAdapter<String> by lazy {

        object : BaseCommonAdapter<String>(mArrayList, R.layout.activity_click_item) {

            override fun onNext(holder: RecyclerHolder, model: String, position: Int) {

                holder!!.setOnClickListener(View.OnClickListener {
                    Toast.makeText(applicationContext, "点击第 " + position + " 个item", Toast.LENGTH_SHORT).show()
                }, R.id.click_text1)

                holder!!.setOnClickListener(View.OnClickListener {
                    Toast.makeText(applicationContext, "点击第 " + position + " 个item - child", Toast.LENGTH_SHORT).show()
                }, R.id.click_text2)
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_click)

        for (i in 0 until 6) {
            mArrayList?.add(i.toString())
        }

        mRecyclerView!!.addItemDecoration(mSpaceDecoration)
        mRecyclerView!!.layoutManager = mLayoutManager
        mRecyclerView!!.adapter = mLoadAdapter
    }
}